texar/modules/pretrained/xlnet_utils.py

```diff
@@ -0,0 +1,164 @@
+    'xlnet-base-cased': _XLNET_PATH + "cased_L-12_H-768_A-12.zip"
+}
+
+
+def _get_assignment_map_from_checkpoint(tvars,  # noqa: C901
+                                        init_checkpoint, scope_name):
+    """
+    Compute the union of the current variables and checkpoint variables.
+    Because of the variable scope of the original XLNet and Texar
+    implementation, we need to build a assignment map to match the variables.
+    """
+    assignment_map = {}
+    initialized_variable_names = {}
+
+    name_to_variable = collections.OrderedDict()
+    for var in tvars:
+        name = var.name
+        m = re.match("^(.*):\\d+$", name)
+        if m is not None:
+            name = m.group(1)
+        name_to_variable[name] = var
+
+    init_vars = tf.train.list_variables(init_checkpoint)
+
+    for check_name, _ in init_vars:
+        check_name_scope = check_name.replace(
+            'model/transformer/', scope_name + '/')
+        model_name = check_name_scope
+        if check_name.startswith('model/lm_loss/bias'):
+            model_name = scope_name + '/lm_loss/bias'
+        elif check_name.startswith('model/transformer/mask_emb'):
+            model_name = check_name_scope.replace(
+                'mask_emb/mask_emb', 'mask_emb')
+        elif check_name.startswith('model/transformer/word_embedding'):
+            model_name = scope_name + '/word_embedder/w'
+        elif re.match('model/transformer/r_[r,s,w]_bias', check_name):
+            model_name = check_name_scope
+        elif re.match('model/transformer/seg_embed', check_name):
+            model_name = check_name_scope
+        elif re.match('model/transformer/layer_\\d+/rel_attn/[q,k,v,r,o]',
+                      check_name):
+            model_name = check_name_scope
+        elif re.match('model/transformer/layer_\\d+/rel_attn/LayerNorm',
+                      check_name):
+            model_name = check_name_scope.replace('LayerNorm/', '')
+        elif re.match('model/transformer/layer_\\d+/ff/layer_[1,2]',
+                      check_name):
+            model_name = check_name_scope.replace('ff/layer_1', 'ff/dense')
+            if model_name == check_name_scope:
+                model_name = check_name_scope.replace(
+                    'ff/layer_2', 'ff/dense_1')
+        elif re.match('model/transformer/layer_\\d+/ff/LayerNorm', check_name):
+            model_name = check_name_scope.replace('LayerNorm/', '')
+
+        if model_name in name_to_variable.keys():
+            assignment_map[check_name] = model_name
+            initialized_variable_names[model_name] = 1
+            initialized_variable_names[model_name + ":0"] = 1
+        else:
+            tf.logging.info('model name:{} not exist'.format(model_name))
+
+    return assignment_map, initialized_variable_names
+
+
+def init_xlnet_checkpoint(init_checkpoint_dir, scope_name):
+    """
+    Initializes XLnet model parameters from a checkpoint.
```

Comment:
"Initializes" or "Initialize"? Keep consistent w/ other docstrings

Reply:
Done in 056c949

Reply:
Fixed in 8a99af3